src/Client.ts

```diff
@@ -279,7 +297,22 @@ export default class Client {
   }
 
   private async sendWakuMessage(msg: WakuMessage): Promise<void> {
-    const ack = await this.waku.lightPush.push(msg)
+    // Waku randomly selects a peer from the Peerstore to send the message to. To ensure this is the
+    // same peer to which we authenticated to, a random peer is selected at this context and then
```

Comment:
Maybe a dumb question but why does it need to be the same peer/node?

Reply:
Authentication state is not replicated between nodes. So if a client authenticates with Node0 but sends the message to Node1 then the message will fail. 


I'll update the comment with more detail